src/codegen/mod.rs

```diff
@@ -2105,6 +2115,19 @@ impl CodeGenerator for CompInfo {
         });
         result.push(tokens);
 
+        if needs_packed_wrapper {
+            let attributes = attributes::derives(&derives);
+            let align = proc_macro2::TokenStream::from_str(
+                &explicit_align.unwrap().to_string(),
+            )
+            .unwrap();
+            result.push(quote! {
+                #attributes
+                #[repr(C, align(#align))]
```

Comment:
Probably better if it'd be `#[repr(transparent)]`, that'd match the ABI exactly, though I don't think it would matter in practice.

Reply:
It doesn't work. we need `align` here but transparent seems doesn't accept other things.

```
error[E0692]: transparent struct cannot have other repr hints
  --> tests/packed-align-conflict.rs:16:8
   |
16 | #[repr(transparent, align(2))]
   |        
```